src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -853,7 +853,7 @@ public final void markDirty()
      */
     public void registerBlockPosition(@NotNull Block block, @NotNull final BlockPos pos)
     {
-        if (block instanceof BlockContainer)
+        if (block instanceof BlockContainer || block instanceof BlockMinecoloniesRack)
```

Comment:
Ehm their is a method in the inventoryutils that gets you all of them (which is what we use to extract anyway). Call that, check if it returns an empty list